File: BineClient/Sdk/bine-sdk/src/test/java/com/msr/bine_sdk/old/HubManagerTest.java
package com.msr.bine_sdk.old;

import android.content.Context;

import com.msr.bine_sdk.secure.BineSharedPreference;

import static org.mockito.Mockito.mock;

import static junit.framework.TestCase.assertEquals;
import static junit.framework.TestCase.assertTrue;
import static junit.framework.TestCase.fail;


/**
 * Example local unit test, which will execute on the development machine (host).
 *
 */
//@RunWith(RobolectricTestRunner.class)
public class HubManagerTest {

    //private Context context = ApplicationProvider.getApplicationContext();
    static final int ASYNC_WAIT = 10;
    final BineSharedPreference sharedPrefs = mock(BineSharedPreference.class);

   /* @Test
    public void listRootData() throws InterruptedException {
//        NetworkService networkService = new MockNetworkService();
//        final CountDownLatch latch = new CountDownLatch(1);
        *//*ContentReceiveCallback callback = new ContentReceiveCallback() {
            @Override
            public void onReceiveMetadata(List<Folder> folders) {
                //TODO: Verify response
                assertEquals("Folders received One", folders.size(), 1);
                latch.countDown();
            }

            @Override
            public void onFailure(Exception exception) {
                //TestCase.fail(exception.getLocalizedMessage());
                exception.printStackTrace();
                latch.countDown();
            }
        };
        ContentRequest request = new ContentRequest("", "", "MSR", "root");
        new HubManager(context).getContent(networkService, getDummyHub(),request ,callback);
        latch.await(ASYNC_WAIT, TimeUnit.SECONDS);*//*
    }
*/
    /*@Test
    public void listContentData() throws InterruptedException {
        /*NetworkService networkService = new MockNetworkService();
        final CountDownLatch latch = new CountDownLatch(1);
        ContentReceiveCallback callback = new ContentReceiveCallback() {
            @Override
            public void onReceiveMetadata(List<Folder> folders) {
                assertEquals("Folders received for ars-talks", folders.size(), 1);
                latch.countDown();
                //TODO: Verify response
            }

            @Override
            public void onFailure(Exception exception) {
                latch.countDown();
            }
        };
        ContentRequest request = new ContentRequest("", "", "MSR", "root");
        new HubManager(context).getContent(networkService, getDummyHub(),request ,callback);
        latch.await(ASYNC_WAIT, TimeUnit.SECONDS);*/
    //}*/

    /*@Test
    public void listOneFolder() throws InterruptedException {
        /*NetworkService networkService = new MockNetworkService();
        final CountDownLatch latch = new CountDownLatch(1);
        ContentReceiveCallback callback = new ContentReceiveCallback() {
            @Override
            public void onReceiveMetadata(List<Folder> folders) {
                assertEquals("Folder received on list one", folders.size(), 1);
                assertEquals(folders.get(0).ID, "ars-talks-ARS 2020 Fairness Transparency and Privacy in Digital Systems");
                latch.countDown();
                //TODO: Verify response
            }

            @Override
            public void onFailure(Exception exception) {
                exception.printStackTrace();
                latch.countDown();
            }
        };
        new HubManager(context).listOne(networkService, getDummyHub(),"MSR", "ars-talks-ARS 2020 Fairness Transparency and Privacy in Digital Systems",callback);
        latch.await(ASYNC_WAIT, TimeUnit.SECONDS);*/
//    }*/
}

File: BineClient/Sdk/bine-sdk/src/test/java/com/msr/bine_sdk/HubManagerTest.kt
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

package com.msr.bine_sdk

import android.net.wifi.WifiConfiguration
import android.net.wifi.WifiInfo
import android.net.wifi.WifiManager
import com.msr.bine_sdk.base.BaseTest
import com.msr.bine_sdk.hub.HubConnectionCallback
import com.msr.bine_sdk.hub.DeviceConnect
import com.msr.bine_sdk.models.Hub
import com.msr.bine_sdk.secure.BineSharedPreference
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import junit.framework.TestCase
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito
import org.robolectric.RobolectricTestRunner
import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

@RunWith(RobolectricTestRunner::class)
class HubManagerTest: BaseTest() {

    val wifiManager: WifiManager = mock()

    private val sharedPrefs: BineSharedPreference = Mockito.mock(BineSharedPreference::class.java)

    @Before
    fun setUpTest() {
        super.setUp()
    }

    @After
    fun setTearDown() = super.tearDown()

    @Test
    fun wifiConnected() {
        val hubManager =
            DeviceConnect(context, sharedPrefs)
        hubManager.setWifiManager(wifiManager)

        whenever(wifiManager.isWifiEnabled)
                .thenReturn(false)

        var isConnected = hubManager.isConnected(getDummyHub())
        TestCase.assertNull(isConnected)

        whenever(wifiManager.isWifiEnabled)
                .thenReturn(true)
        val wifiInfo:WifiInfo = mock()
        whenever(wifiInfo.ssid).thenReturn("\"JioFi3_566200\"")
        whenever(wifiManager.connectionInfo).thenReturn(wifiInfo)

        isConnected = hubManager.isConnected(getDummyHub())
        TestCase.assertEquals("Wifi config empty", "JioFi3_566200", isConnected.SSID)
        TestCase.assertEquals("Wifi config empty", "192.168.2.2", isConnected.ip)
    }

    @Test
    @Throws(InterruptedException::class)
    fun connect() {
        val wifiManager:WifiManager = mock()
        val hubManager =
            DeviceConnect(context, sharedPrefs)

        hubManager.setWifiManager(wifiManager)
        whenever(wifiManager.isWifiEnabled).thenReturn(true)
        val wifiConfig = WifiConfiguration()
        wifiConfig.SSID = "\"JioFi3_566200\""
        wifiConfig.preSharedKey = "\"kc78k1x8zb\""
        whenever(wifiManager.addNetwork(wifiConfig)).thenReturn(2)
        whenever(wifiManager.enableNetwork(2, true)).thenReturn(true)
        whenever(wifiManager.disconnect()).thenReturn(true)
        whenever(wifiManager.reconnect()).thenReturn(true)

        val latch = CountDownLatch(1)
        hubManager.connect(getDummyHub(), object : HubConnectionCallback {
            override fun onConnectFailure(response: Hub, exception: Exception) {
                TestCase.assertTrue("Wifi connect", true)
                latch.countDown()
            }

            override fun onConnect(hub: Hub) {
                TestCase.assertTrue("Wifi connect", true)
                latch.countDown()
            }
        })
        latch.await(ASYNC_WAIT.toLong(), TimeUnit.SECONDS)
    }

    private fun getDummyHub(): Hub? {
        val hub = Hub()
        hub.name = "JioFi3_566200"
        hub.wifi5GPass = "kc78k1x8zb"
        hub.ip = "192.168.2.2"
        hub.wifi5GSSID = "JioFi3_566200"
        return hub
    }
}